### PR TITLE
Add tabIndex property for Button

### LINKED ISSUE #6200: Button component doesn't have tabIndex prop
## Environment
1. `npx react-native --version`: 4.13.0
2. `npx react-native run-windows --info`:
```
  System:
    OS: Windows 10 10.0.19041
    CPU: (40) x64 Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz
    Memory: 157.00 GB / 190.60 GB
  Binaries:
    Node: 12.18.3 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.17.0-20190429.1820 - C:\open\fbsource\xplat\third-party\yarn\yarn.BAT
    npm: 6.14.6 - C:\Program Files\nodejs\npm.CMD
    Watchman: 20200724.030019 - C:\tools\watchman\watchman.EXE
  Installed UWP SDKs:
    10.0.18362.0
```
3. `reg query "HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\AppModelUnlock"`
```
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\AppModelUnlock
    AllowDevelopmentWithoutDevLicense    REG_DWORD    0x1
    AllowAllTrustedApps    REG_DWORD    0x1
```

## Steps To Reproduce
1. Create app with following JSX:
```
      <Button title="button 1" {...{tabIndex: 1}} />
      <Button title="button 3" {...{tabIndex: 3}} />
      <Button title="button 2" {...{tabIndex: 2}} />

```
2. Run it and tab through the buttons

## Expected Results
Expected tab order: button 1, button 2, button 3
Actual tab order: button 1, button 3, button 2

This is because https://github.com/microsoft/react-native-windows/blob/1f2049d2dfc6f3398f71bb176e14911efb780643/vnext/src/Libraries/Components/Button.windows.js doesn't pass `tabIndex` to `Touchable`.

### PR DESCRIPTION
**Pull request by rectified95**
Fixes #6200 

The tabIndex prop needs to be forwarded to the underlying TouchableHighlight in order to make its way to ViewViewManager, which implements it. 
Also had to change the JSValueType that is expected from Double to Int64 for it to work.

Tested locally running:
```
<Button title="button 1" {...{tabIndex: 1}} />
<Button title="button 3" {...{tabIndex: 3}} />
<Button title="button 2" {...{tabIndex: 2}} />
```
buttonIndex

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6732)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Views/ViewViewManager.cpp [hunk 1]
```diff
@@ -418,7 +418,7 @@ bool ViewViewManager::UpdateProperty(
       else if (propertyValue.IsNull())
         pViewShadowNode->EnableFocusRing(false);
     } else if (propertyName == "tabIndex") {
-      if (propertyValue.Type() == winrt::Microsoft::ReactNative::JSValueType::Double) {
+      if (propertyValue.Type() == winrt::Microsoft::ReactNative::JSValueType::Int64) {
         auto tabIndex = propertyValue.AsDouble();
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/Views/ViewViewManager.cpp:422]: seems wrong to do AsDouble here if the property is an int64.
You can just do AsInt64 and remove the property type check too

**Comment by NickGerleman** [comment]
I remember Liron hit this during prototyping, but did we ever settle on whether we wanted to keep tabIndex long term? Last I knew that was still an open question.

If we didn’t want it long term, it may be better to avoid adding support in more places. We don’t support tabIndex for some other components either like Pressable.

File: vnext/Microsoft.ReactNative/Views/ViewViewManager.cpp (hunk 1 above)
Comment (NickGerleman) [reply vnext/Microsoft.ReactNative/Views/ViewViewManager.cpp:422]: The representation can also be different when web debugging vs not, so the type check can break things there. +1 to just using AsDouble.

File: vnext/src/Libraries/Components/Button.windows.js [hunk 2]
```diff
@@ -135,6 +135,11 @@ type ButtonProps = $ReadOnly<{|
     Used to locate this view in end-to-end tests.
    */
   testID?: ?string,
+
+  /**
+    Set the order in which elements receive focus when the user navigates through them by pressing Tab.
+   */
+  tabIndex?: ?number,
```
Comment (NickGerleman) [inline vnext/src/Libraries/Components/Button.windows.js:142]: Please mark anything Windows specific.

File: vnext/Microsoft.ReactNative/Views/ViewViewManager.cpp (hunk 1 above)
Comment (rectified95) [reply vnext/Microsoft.ReactNative/Views/ViewViewManager.cpp:422]: Will do. 
@NickGerleman Can you explain how web debugging affects this? Also, you wrote ' +1 to AsDouble' but meant AsInt64, as Alex said, right?

**Comment by rectified95** [comment]
We can easily add it to Pressable as well, since it is already in View. Since Chris assigned this issue to me, I assume the decision was made?

**Review by asklar — APPROVED** [review]

File: change/react-native-windows-136ae86b-5757-4cb4-8cec-a50da19528c3.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "Fix tabINdex property for Button",
```
Comment (asklar) [inline change/react-native-windows-136ae86b-5757-4cb4-8cec-a50da19528c3.json:3]:
```suggestion
  "comment": "Fix tabIndex property for Button",
```

**Pull request merged**